src/include/ikarus/Geometries/GeometryType.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+#include <iostream>
+
+// Further types should be added in the future
+namespace Ikarus {
+  enum class GeometryType {
+    vertex,
+    linearLine,
+    quadraticLine,
+    cubicLine,
+    linearTriangle,
+    linearQuadrilateral,
+    linearTetrahedron,
+    pyramid,
+    prism,
+    linearHexahedron
+  };
+
+  inline bool isVertex(GeometryType type) { return type == Ikarus::GeometryType::vertex; }
+  inline bool isLinearLine(GeometryType type) { return type == Ikarus::GeometryType::linearLine; }
+  inline bool isQuadraticLine(GeometryType type) { return type == Ikarus::GeometryType::quadraticLine; }
+  inline bool isCubicLine(GeometryType type) { return type == Ikarus::GeometryType::cubicLine; }
+  inline bool isLinearTriangle(GeometryType type) { return type == Ikarus::GeometryType::linearTriangle; }
+  inline bool isLinearQuadrilateral(GeometryType type) { return type == Ikarus::GeometryType::linearQuadrilateral; }
+  inline bool isLinearTetrahedron(GeometryType type) { return type == Ikarus::GeometryType::linearTetrahedron; }
+  inline bool isPyramid(GeometryType type) { return type == Ikarus::GeometryType::pyramid; }
+  inline bool isPrism(GeometryType type) { return type == Ikarus::GeometryType::prism; }
+  inline bool isLinearHexahedron(GeometryType type) { return type == Ikarus::GeometryType::linearHexahedron; }
+
+  inline int dimension(GeometryType type) {
+    if (isVertex(type)) {
+      return 0;

```

Comment:
We should at to the current test some lines to also test for the untested lines. Furthermore, should that not be a switch statement?

I would argue the following is more readable
```cpp
  inline int dimension(GeometryType type) {
    switch (type) {
      case GeometryType::vertex:
        return 0;
      case GeometryType::linearLine:
      case GeometryType::quadraticLine:
      case GeometryType::cubicLine:
        return 1;
      case GeometryType::linearTriangle:
      case GeometryType::linearQuadrilateral:
        return 2;
      case GeometryType::linearTetrahedron:
      case GeometryType::pyramid:
      case GeometryType::prism:
      case GeometryType::linearHexahedron:
        return 3;
      default:
        throw std::runtime_error(
            "Dimension of the given type couldn't be obtained. Check function 'dimension' in GeometryType.h");
    }
}
```